### PR TITLE
Rebase on `curated-transformers` 2.0.0

### PR DESCRIPTION
**Pull request by shadeMe**
## Description

This PR rebases the project on `curated-transformers` 1.0.0. This is a breaking change - spaCy pipelines trained with previous versions will not be compatible due to changes to parameter names, etc. 

The piece encoder/tokenizer implementations remain untouched (except for a minor fix) for the time being since the serialization contract is currently undetermined on the `curated-transformers` side.

### Types of change

Maintenance

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

**Comment by adrianeboyd** [comment]
I'm worried that this may cause the same kinds of headaches as `spacy-transformers` v1.3.0. The only info we have in saved pipeline directories is which spacy version they were trained with, the versioning for `spacy-transformers` or `spacy-curated-transformers` is only included by `spacy package`.

**Comment by adrianeboyd** [comment]
It's nothing against this PR, just potentially a concern about publishing anything that uses earlier versions, such as the `trf` pipelines that have been published for spacy v3.7.0.dev0.

**Comment by danieldk** [comment]
I have already reviewed @shadeMe's commits, so only mine still need reviewing:

https://github.com/explosion/spacy-curated-transformers/pull/19/files/25d87e1d46926f86b578cdc9a7a38699c71bfd52..b9c93d72c502b313b01f2e6ab4d0e0541e61adf7

File: spacy_curated_transformers/models/architectures.py [hunk 1]
```diff
@@ -806,16 +807,35 @@ def build_pytorch_checkpoint_loader_v1(*, path: Path) -> Callable[
     path (Path):
         Path to the PyTorch checkpoint.
     """
+    return build_pytorch_checkpoint_loader_v2(path=path.parent)
```
Comment (shadeMe) [inline spacy_curated_transformers/models/architectures.py:810]: Think we can also update the v1 docstring to mention the support for SafeTensors.

**Review by shadeMe — COMMENTED** [review]
LGTM! Just one small comment.

File: spacy_curated_transformers/models/architectures.py (hunk 1 above)
Comment (danieldk) [reply spacy_curated_transformers/models/architectures.py:810]: Added.

**Pull request merged**